cirq/google/engine/engine_sampler.py

```diff
@@ -61,6 +61,34 @@ def run_sweep(
                                          gate_set=self._gate_set)
         return job.results()
 
+    def run_batch(
+            self,
+            programs: List['cirq.Circuit'],
+            params_list: Optional[List['cirq.Sweepable']] = None,
+            repetitions: Union[int, List[int]] = 1,
+    ) -> List[List['cirq.TrialResult']]:
+        """Runs the supplied circuits.
+
+        In order to gain a speedup from using this method instead of other run
+        methods, the number of circuit repetitions must be the same for all
+        circuits. That is, the `repetitions` argument must be an integer, or
+        else a list with identical values.
+        """
+        if isinstance(repetitions, List) and len(programs) != len(repetitions):
+            raise ValueError('Number of circuits and repetitions must match')
```

Comment:
Maybe put "Number of circuits and repetitions specified must match." so that people don't get confused between the number of reps in each circuit and the number of elements in the repetition list?  I am not sure how to make that more clear.

Reply:
I've made the messages more clear.